src/vcon/MarketGovernance.sol

```diff
@@ -0,0 +1,650 @@
+
+    /// ---------- Per Venue User Profit Tracking ----------
+
+    /// @dev convention for all double nested address mappings is key (venue -> user) -> value
+
+    /// @notice record how much VCON a user deposited in a given venue
+    mapping(address => mapping(address => uint256)) public venueUserShares;
+
+    /// @param _core reference to core
+    /// @param _pcvRouter reference to pcvRouter
+    constructor(address _core, address _pcvRouter) CoreRefV2(_core) {
+        pcvRouter = _pcvRouter;
+    }
+
+    /// @notice update the VCON share price and the last recorded profit for a given venue
+    /// @param venue address to accrue
+    function accrueVcon(address venue) external globalLock(1) whenNotPaused {
+        IPCVOracle oracle = pcvOracle();
+
+        require(oracle.isVenue(venue), "MarketGovernance: invalid destination");
+
+        _accrue(venue);
+    }
+
+    /// ---------- Permissionless User PCV Allocation Methods ----------
+
+    /// @notice a user can get slashed up to their full VCON stake for entering
+    /// a venue that takes a loss.
+    /// @param amountVcon to stake on destination
+    /// @param venue address to accrue rewards to, and send funds to
+    function stake(
+        uint256 amountVcon,
+        address venue
+    ) external globalLock(1) whenNotPaused {
+        IPCVOracle oracle = pcvOracle();
+        require(oracle.isVenue(venue), "MarketGovernance: invalid destination");
+
+        _accrue(venue); /// update share price in the destination so the user gets in at the current share price
+
+        /// vconToShares will always return correctly as accrue will set venueLastRecordedVconSharePrice
+        /// to the correct share price from 0 if uninitialized
+        uint256 userShareAmount = vconToShares(venue, amountVcon);
+
+        /// user updates
+        venueUserShares[venue][msg.sender] += userShareAmount;
+
+        /// venue updates
+        venueTotalShares[venue] += userShareAmount;
+
+        /// check and an interaction with a trusted contract
+        vcon().safeTransferFrom(msg.sender, address(this), amountVcon); /// transfer VCON in
+
+        emit Staked(venue, msg.sender, amountVcon);
+    }
+
+    /// @notice unstake VCON and transfer corresponding VCON to another venue
+    /// @param shareAmount the amount of shares to unstake
+    /// @param venue address to accrue rewards to, and pull funds from
+    /// @param vconRecipient address to receive the VCON
+    /// @dev both venue and destination are checked twice,
+    /// the first time in the market governance contract,
+    /// the second time in the PCV Router contract.
+    function unstake(
+        uint256 shareAmount,
+        address venue,
+        address vconRecipient
+    ) external globalLock(1) whenNotPaused {
+        /// ---------- Checks ----------
+        IPCVOracle oracle = pcvOracle();
+
+        require(oracle.isVenue(venue), "MarketGovernance: invalid venue");
+        require(
```

Comment:
This check must be removed (and also from `applyVenueLosses`): if a venue is deprecated, governance will withdraw all funds from there & send to the holding deposit, and then unlist the empty deprecated venue from the PCVOracle, so VCON users that previously allocated to the deprecated venue won't be able to unstake